### PR TITLE
add redirect privacy-policy and cookie-declaration to pubpub 

### PR DESCRIPTION
**Pull request by ali-tm-amin**
The continuumtest proxy is set on builder-configurations https://github.com/elifesciences/builder-configuration/commit/ad83e8d22b655e068707159cb54c7ae65494e321

### CODE REVIEW

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 1]
```diff
@@ -325,8 +325,30 @@ server {
...
+    location ~ ^/privacy-policy[/]* {
+        proxy_pass {{ pillar.journal.privacy_notice_url }};
+        proxy_ssl_server_name on;
+        proxy_connect_timeout 10s;
+        proxy_read_timeout 10s;
+        proxy_set_header X-PubPubClient elifesciences;
+    }
+    {% endif -%}
+
+     {% if pillar.journal.cookie-declaration %}
```
Comment (scottaubrey) [inline salt/journal/config/etc-nginx-sites-available-journal.conf:339]: Should this be `cookie_declaration_url`? Also indentation looks a little off.

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 2]
```diff
@@ -325,8 +325,30 @@ server {
     }
     {% endif -%}
 
+    {% if pillar.journal.privacy-policy %}
```
Comment (scottaubrey) [inline salt/journal/config/etc-nginx-sites-available-journal.conf:328]: Should this be `privacy_potice_url`?

**Review by scottaubrey — CHANGES_REQUESTED** [review]
I think the logic statement values look a little off and might need tweaking a little, but I could be wrong.

File: salt/journal/config/etc-nginx-sites-available-journal.conf (hunk 1 above)
Comment (ali-tm-amin) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Thanks, I've corrected it and removed the empty space

File: salt/journal/config/etc-nginx-sites-available-journal.conf (hunk 2 above)
Comment (ali-tm-amin) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: This is changed to `privacy_notice_url`

File: salt/journal/config/etc-nginx-sites-available-journal.conf [hunk 3]
```diff
@@ -325,8 +325,29 @@ server {
     }
     {% endif -%}
 
+    {% if pillar.journal.privacy_policy_url %}
```
Comment (scottaubrey) [inline salt/journal/config/etc-nginx-sites-available-journal.conf:328]: Ah! seen it: should be `privacy_notice_url`

**Review by scottaubrey — CHANGES_REQUESTED** [review]

File: salt/journal/config/etc-nginx-sites-available-journal.conf (hunk 3 above)
Comment (ali-tm-amin) [reply salt/journal/config/etc-nginx-sites-available-journal.conf]: Oh dear, so it wasn't the first letter I typed mistakenly.

**Comment by scottaubrey** [comment]
A spent a bit of time recreating the issue (because I need to do it outside of builder and the instance is destroyed after the CI run). 

The error is: 
```
2024/09/09 16:39:59 [emerg] 1309#1309: "proxy_pass" cannot have URI part in location given by regular expression, or inside named location, or inside "if" statement, or inside "limit_except" block in /etc/nginx/sites-enabled/journal.conf:332
```

According to https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass we can't do what we're doing.

So, I suggest a simpler approach: just do a simple location match. I'll push it up and see what you think.

**Comment by ali-tm-amin** [comment]
> A spent a bit of time recreating the issue (because I need to do it outside of builder and the instance is destroyed after the CI run).
> 
> The error is:
> 
> ```
> 2024/09/09 16:39:59 [emerg] 1309#1309: "proxy_pass" cannot have URI part in location given by regular expression, or inside named location, or inside "if" statement, or inside "limit_except" block in /etc/nginx/sites-enabled/journal.conf:332
> ```
> 
> According to https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass we can't do what we're doing.
> 
> So, I suggest a simpler approach: just do a simple location match. I'll push it up and see what you think.

That's great thanks a lot

**Comment by scottaubrey** [comment]
Are you wanting to merge this in now then? @ali-tm-amin

**Review by scottaubrey — APPROVED** [review]

**Comment by ali-tm-amin** [comment]
> https://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_pass

Yes, I will do it now, thanks

**Pull request merged**